cmds/chroot/chroot.go

```diff
@@ -0,0 +1,191 @@
+package main
+
+// +build unix
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"syscall"
+)
+
+var (
+	userspecFlag  string
+	groupsFlag    string
+	skipchdirFlag bool
+)
+
+func init() {
+	flag.StringVar(&userspecFlag, "u", "", "specify user and group (ID only) as USER:GROUP")
```

Comment:
One thing you could do with this flag and the groups is write a `flag.Value` implementation that does the parsing just as you do below. So this would become

```go
type userSpec struct {
  uid uint32
  gid uint32
}

// Set implements flag.Value.Set.
func (u *userSpec) Set(s string) error {
  var err error
  // You should inline this, no need to keep parseUserspec around if you have the flag.Value.
  u.uid, u.gid, err = parseUserspec(s)
  return err
}

// Get implements flag.Getter.Get.
func (u *userSpec) Get() interface{} {
  return *u
}

// String implements flag.Value.String.
func (u *userSpec) String() string {
  return fmt.Sprintf("%d:%d", u.uid, u.gid)
}

var (
  // the value this is set to is the flag's default value.
  user userSpec = defaultUser()
)

func defaultUser() userSpec {
  return userSpec{
    uid: uint32(os.Getuid()),
    gid: uint32(os.Getgid()),
  }
}

func init() {
  flag.Var(&user, "u", "specify user and group (ID only) as user:group")
}
```

`flag.Parse` will then take care of all the appropriate error handling.